students/ethan_nguyen/lesson09/mailroom9.py

```diff
@@ -0,0 +1,104 @@
+
+import sys, os
+import pytest
+
+'''
+Class to store Donor object and its attributes
+'''
+class Donor:
+    def __init__(self, donor_name, amount=0):
+        self._name = donor_name
+        self._amount_gift = amount
+        self.count_donation = 1
+    
+    def __str__(self):
+        return str(f"Donor: {self.name}")
+
+    def __repr__(self):
+        return repr(f'Donor({self.name})')
+
+    '''function to return historical amount'''
+    @property
+    def amount(self):
+        return self._amount_gift
+    @amount.setter
+    def amount(self, value):
+        self._amount_gift = value
+
+    '''function to return donor name'''
+    @property
+    def name(self):
+        return self._name
+    @name.setter
+    def name(self, value):
+        self._name = value
+
+    def sort_key(self):
+        return self.amount
+
+    '''function to calculate average donation'''
+    def cal_average(self):
+        return (self.amount/self.count_donation)
+
+    def __eq__(self, other):
+        return (self._amount_gift == other._amount_gift)
+    
+    def __lt__(self, other):
+        return (self._amount_gift < other._amount_gift)
+
+    def __gt__(self, other):
+        return (self._amount_gift > other._amount_gift)
+
+    '''function to return the thank you string'''
+    def create_thank_you_note(self):
+        return f'Dear {self.name}, \n Thank you for your very kind donation of ${self.amount:,.2f} \n \
+            It will be put to very good use. \n Sincerely, \n -UW'
+
+class DonorCollection(dict):
+    def __init__(self):
+        super().__init__(self)
+    
+    def is_donor_exist(self, Donor):
+        if Donor.name in self:
+            return True
```

Comment:
this really confused me reading at first - `Donor` is a class name, `donor` is a variable 

this should be `donor`

Reply:
this can be simplified `return donor.name in self`